services/surveys.js

```diff
@@ -1,53 +1,95 @@
-const { Survey, SurveyChoice, SurveyResponse } = require('../models');
+const _ = require('lodash');
+const moment = require('moment');
+
+const { SurveyResponse } = require('../models');
+const contentApi = require('./content-api');
 
 function findAll() {
-    return Survey.findAll({
-        include: [
-            {
-                model: SurveyChoice,
-                as: 'choices',
-                required: true,
-                include: [
-                    {
-                        model: SurveyResponse,
-                        as: 'responses',
-                        required: true
-                    }
-                ]
-            }
-        ]
+    // fetch all surveys (expired and live) from API
+    const getSurveys = contentApi.getSurveys({
+        showAll: true
     });
-}
 
-function findActiveWithChoices({ filterByPath }) {
-    return Survey.findAll({
-        where: {
-            active: true
-        },
-        include: [
-            {
-                model: SurveyChoice,
-                as: 'choices',
-                required: true
-            }
-        ]
-    }).then(surveys => {
-        if (filterByPath) {
-            return surveys.filter(s => s.activePath === filterByPath);
-        } else {
-            return surveys;
-        }
-    });
-}
+    // get user responses from the database
+    const getResponses = SurveyResponse.findAll();
+    const normalisedDateFormat = 'YYYY-MM-DD';
+
+    // combine the votes with the choices
+    return Promise.all([getSurveys, getResponses]).then(responses => {
+        let surveys = responses[0];
+        let votes = responses[1];
+
+        // merge the datasets
+        let mergedSurveys = surveys.map(survey => {
+
+            // append responses to the relevant choice
+            survey.choices = survey.choices.map(choice => {
+
+                // retrieve the votes for this survey's choices
+                choice.responses = votes.filter(v => {
+                    return v.survey_id === survey.id && v.choice_id === choice.id;
+                });
+
+                // fill in gaps for days without votes
+                let oldestVote = _.minBy(choice.responses, 'createdAt');
+                let newestVote = _.maxBy(choice.responses, 'createdAt');
+
+                // calculate votes per day (or substitute with zeroes)
+                if (newestVote && oldestVote) {
+                    let daysInRange = moment(newestVote.createdAt).diff(moment(oldestVote.createdAt), 'days');
+
+                    let counts = {};
+                    // group votes by day to aid in graphing
+                    choice.responses.forEach(response => {
```

Comment:
Could do this with `reduce` rather than `forEach` to make it into one statement?